### PR TITLE
Advance numba from 0.36.1 to 0.40.0

### PR DESCRIPTION
**Pull request by Naman-Priyadarshi**
- [x] Closes https://github.com/pvlib/pvlib-python/issues/1439
- [x]  I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
- [x] Adds description and name entries in the appropriate "what's new" file in [docs/sphinx/source/whatsnew](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with :issue:`num` or this Pull Request with :pull:`num`. Includes contributor name and/or GitHub username (link with :ghuser:`user`).
- [x] Pull request is nearly complete and ready for detailed review.

Updated numba version from 0.36.1 to 0.40.0 in asv.conf

### CODE REVIEW

**Comment by Naman-Priyadarshi** [comment]
Oops Sorry! looks like I named the whatsnew.rst commit same as the previous one.

File: docs/sphinx/source/whatsnew/v0.9.2.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ Bug fixes
 * :py:func:`pvlib.irradiance.get_total_irradiance` and
   :py:func:`pvlib.solarposition.spa_python` now raise an error instead
   of silently ignoring unknown parameters (:ghpull:`1437`)
-  
+* Updated version of numba in asv.conf from 0.36.1 to 0.40.0 to solve numba/numpy conflict. (:issue:`1439`, :pull:`1440`)
```
Comment (kanderso-nrel) [inline docs/sphinx/source/whatsnew/v0.9.2.rst:17]: Let's move this to a new `Benchmarking` section instead of `Bug fixes`

Comment (Naman-Priyadarshi) [reply docs/sphinx/source/whatsnew/v0.9.2.rst:17]: Sound good! I'll update it

**Review by wholmgren — APPROVED** [review]
No objection, but if there is a problem then maybe it's worth enforcing in setup.py? I'm also fine with moving to 3.7 and whatever the corresponding numba ends up being, but that is a substantially larger scope than this PR currently presents.

**Comment by kanderso-nrel** [comment]
Thanks @wholmgren.  As I understand it the problem is that conda on windows refuses to install the specific combination of `numba==0.36.1` and `numpy==1.16.0`.  I can't think of why that combination would appear in practice unless you're shooting for the minimum environment like we are here.  So I'm inclined to merge this and kick any setup.py and other benchmark environment update cans down the road.

Thanks again @Naman-Priyadarshi!

**Pull request merged**